echopype/commongrid/api.py

```diff
@@ -214,5 +221,108 @@ def compute_MVBS_index_binning(ds_Sv, range_sample_num=100, ping_num=100):
+    This is different from Echoview's assumption of constant ping rate, vessel speed, and sample
+    thickness when computing mean Sv.
+    """
+    # Check Sv contains lat/lon
+    if "latitude" not in ds_Sv or "longitude" not in ds_Sv:
+        raise ValueError("Both 'latitude' and 'longitude' must exist in the input Sv dataset.")
+
+    # Check if depth vectors are identical within each channel
+    if not ds_Sv["depth"].groupby("channel").map(check_identical_depth).all():
+        raise ValueError(
+            "Only Sv data with identical depth vectors across all pings "
+            "are allowed in the current compute_NASC implementation."
+        )
+
+    # Get distance from lat/lon in nautical miles
+    dist_nmi = get_distance_from_latlon(ds_Sv)
+
+    # Find binning indices along distance
+    bin_num_dist, dist_bin_idx = get_dist_bin_info(dist_nmi, cell_dist)  # dist_bin_idx is 1-based
+
+    # Find binning indices along depth: channel-dependent
+    bin_num_depth, depth_bin_idx = get_depth_bin_info(ds_Sv, cell_depth)  # depth_bin_idx is 1-based
+
+    # Compute mean sv (volume backscattering coefficient, linear scale)
+    # This is essentially to compute MVBS over a the cell defined here,
+    # which are typically larger than those used for MVBS.
+    # The implementation below is brute force looping, but can be optimized
+    # by experimenting with different delayed schemes.
+    # The optimized routines can then be used here and
+    # in commongrid.compute_MVBS and clean.estimate_noise
+    sv_mean = []
+    for ch_seq in np.arange(ds_Sv["channel"].size):
+        # TODO: .compute each channel sequentially?
+        #       dask.delay within each channel?
+        ds_Sv_ch = ds_Sv["Sv"].isel(channel=ch_seq).data  # preserve the underlying type
+
+        sv_mean_dist_depth = []
+        for dist_idx in np.arange(bin_num_dist) + 1:  # along ping_time
+            sv_mean_depth = []
+            for depth_idx in np.arange(bin_num_depth) + 1:  # along depth
+                # Sv dim: ping_time x depth
+                Sv_cut = ds_Sv_ch[dist_idx == dist_bin_idx, :][
+                    :, depth_idx == depth_bin_idx[ch_seq]
+                ]
+                sv_mean_depth.append(np.nanmean(10 ** (Sv_cut / 10)))
+            sv_mean_dist_depth.append(sv_mean_depth)
+
+        sv_mean.append(sv_mean_dist_depth)
+
+    # Compute mean height
+    # For data with uniform depth step size, mean height = vertical size of cell
+    height_mean = cell_depth
+    # TODO: generalize to variable depth step size
+
+    ds_NASC = xr.DataArray(
+        np.array(sv_mean) * height_mean,
+        dims=["channel", "distance", "depth"],
+        coords={
+            "channel": ds_Sv["channel"].values,
+            "distance": np.arange(bin_num_dist) * cell_dist,
+            "depth": np.arange(bin_num_depth) * cell_depth,
+        },
+    )
+
+    # TODO: Attach attributes
```

Comment:
Variable (DataArray) attributes, right?

Reply:
Yes, I was thinking something along the line of what we have for `compute_MVBS`.

Reply:
Ah. Thanks for pointing to `compute_MVBS`; that'll make it easier!